client/src/modules/stock/entry/modals/findPurchase.modal.js

```diff
@@ -6,43 +6,55 @@ StockFindPurchaseModalController.$inject = [
   'uiGridConstants', 'GridFilteringService', 'ReceiptModal',
 ];
 
-function StockFindPurchaseModalController(Instance, Purchase, Notify,
-  uiGridConstants, Filtering, Receipts) {
+function StockFindPurchaseModalController(
+  Instance, Purchase, Notify,
+  uiGridConstants, Filtering, Receipts
+) {
   var vm = this;
+  var filtering;
+  var columns;
 
   // global
   vm.selectedRow = {};
 
   /* ======================= Grid configurations ============================ */
   vm.filterEnabled = false;
-  vm.gridOptions = { appScopeProvider: vm };
+  vm.gridOptions = { appScopeProvider : vm };
 
-  var filtering = new Filtering(vm.gridOptions);
+  filtering = new Filtering(vm.gridOptions);
 
-  var columns = [
-    { field            : 'reference',
+  columns = [
+    {
+      field            : 'reference',
       displayName      : 'TABLE.COLUMNS.REFERENCE',
       headerCellFilter : 'translate',
-      cellTemplate     : 'modules/stock/entry/modals/templates/purchase_reference.tmpl.html' },
+      cellTemplate     : 'modules/stock/entry/modals/templates/purchase_reference.tmpl.html',
+    },
+
     {
       field            : 'date',
       cellFilter       : 'date',
```

Comment:
Note: we should add the bhConstants date format parameter here to ensure it renders the same across French/English clients.